### PR TITLE
Feature v0.10.0 telemetry

### PR DESCRIPTION
**Pull request by afeiszli**
Adds telemetry which can be optionally disabled with TELEMETRY="off".
Telemetry sent to PostHog once every 24 hours and sends non-identifying info:
- count of nodes
- types of nodes
- count of networks
- count of users
- version

### CODE REVIEW

File: database/database.go [hunk 1]
```diff
@@ -184,6 +194,25 @@ func FetchRecords(tableName string) (map[string]string, error) {
 	return getCurrentDB()[FETCH_ALL].(func(string) (map[string]string, error))(tableName)
 }
 
+// InitializeUUID - create a UUID record for server if none exists
+func InitializeUUID() error {
```
Comment (0xdcarns) [inline database/database.go:198]: should be private and add to InitializeDatabase func

File: serverctl/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		return err
+	}
+	//set telemetry timestamp for server, restarts 24 hour cycle
+	return setTelemetryTimestamp(serverUUID)
+}
+
+// fetchTelemetry - fetches telemetry data: count of various object types in DB
+func fetchTelemetryData() (TelemetryData, error) {
+	var data TelemetryData
+	extData, _ := database.FetchRecords(database.EXT_CLIENT_TABLE_NAME)
```
[the remark this answers is not in the document]
Comment (afeiszli) [reply serverctl/telemetry.go]: added via new function

File: serverctl/telemetry.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	rawData, err = database.FetchRecord(database.SERVER_UUID_TABLE_NAME, database.SERVER_UUID_RECORD_KEY)
+	if err != nil {
+		return telObj, err
+	}
+	err = json.Unmarshal([]byte(rawData), &telObj)
+	return telObj, err
+}
+
+// TelemetryData - What data to send to posthog
+type TelemetryData struct {
```
[the remark this answers is not in the document]
Comment (afeiszli) [reply serverctl/telemetry.go]: made private

File: database/database.go (hunk 1 above)
Comment (afeiszli) [reply database/database.go]: made private and added to initialize

File: database/database.go [hunk 4]
```diff
@@ -184,6 +195,25 @@ func FetchRecords(tableName string) (map[string]string, error) {
...
+		}
+	} else if len(records) > 0 {
+		return nil
+	}
+	telemetry := models.Telemetry{UUID: uuid.NewString()}
+	telJSON, err := json.Marshal(telemetry)
+	if err != nil {
+		return err
+	}
+	err = Insert(SERVER_UUID_RECORD_KEY, string(telJSON), SERVER_UUID_TABLE_NAME)
```
Comment (0xdcarns) [inline database/database.go:213]: just return Insert(...) here

File: serverctl/telemetry.go [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+	"github.com/gravitl/netmaker/database"
+	"github.com/gravitl/netmaker/logger"
+	"github.com/gravitl/netmaker/logic"
+	"github.com/gravitl/netmaker/models"
+	"github.com/gravitl/netmaker/servercfg"
+	"github.com/posthog/posthog-go"
+)
+
+const POSTHOG_PUB_KEY = "phc_1vEXhPOA1P7HP5jP2dVU9xDTUqXHAelmtravyZ1vvES"
```
Comment (0xdcarns) [inline serverctl/telemetry.go:15]: all public variables/funcs should have comments, i.e.
// POSTHOG_PUB_KEY - is public key to access post hog api

File: serverctl/telemetry.go [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+// getDBLength - get length of DB to get count of objects
+func getDBLength(dbname string) int {
+	data, err := database.FetchRecords(dbname)
+	if err != nil {
+		return 0
+	}
+	return len(data)
+}
+
+// TelemetryData - What data to send to posthog
```
Comment (0xdcarns) [inline serverctl/telemetry.go:157]: lower case these comments or remove since they're private

**Pull request merged**